### PR TITLE
refactor(code): Minor tweaks, refactors and code style changes

### PR DESCRIPTION
**Pull request by tibetiroka**
**Refactor:** This PR does some small (single-line) refactors in the engine code.

## Refactor Details
- Added `explicit` to all single-argument constructors that weren't used implicitly (as recommended by the style guide)
- ~~Added/removed `std::move` as necessary~~
- Added a couple of reference parameters instead of copies
- Replaced some `size()` calls with `empty()`
- Replaced some iterator-based for loops with range-based loops
- Added `const` to a couple functions that could be made const

and some other small things.

## Testing Done
CI.

### Automated Tests Added
N/A

## Performance Impact
N/A. Might be slightly better with `empty()` and `move()`, but probably insignificant.

### CODE REVIEW

**Comment by RisingLeaf** [comment]
all looks good, but why this:
> Replaced some iterator-based for loops with range-based loops

**Comment by tibetiroka** [comment]
From the style guide:
> Prefer range-based for loops whenever possible.
> Range-based for is more clear and concise than using an index or an iterator.

File: source/MenuPanel.cpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ MenuPanel::MenuPanel(PlayerInfo &player, UI &gamePanels)
 		for(const auto &source : GameData::Sources())
 		{
 			auto credit = Format::Split(Files::Read(source + "credits.txt"), "\n");
-			if((credit.size() > 1) || (credit.front() != ""))
+			if(!credit.empty() || credit.front().empty())
```
Comment (quyykk) [inline source/MenuPanel.cpp:69]: not equivalent :smile:

File: source/text/DisplayText.cpp [hunk 2]
```diff
@@ -25,7 +27,7 @@ DisplayText::DisplayText(const char *text, Layout layout)
 
 
 DisplayText::DisplayText(const std::string &text, Layout layout)
-	: layout(layout), text(text)
+	: layout(layout), text(std::move(text))
```
Comment (quyykk) [inline source/text/DisplayText.cpp:30]: `text` is const, so the move never moves.

File: source/MenuPanel.cpp (hunk 1 above)
Comment (tibetiroka) [reply source/MenuPanel.cpp:69]: Oops.

Comment (quyykk) [reply source/MenuPanel.cpp]: @tibetiroka ~~one more~~

Comment (tibetiroka) [reply source/MenuPanel.cpp]: What do you mean?

Comment (tibetiroka) [reply source/MenuPanel.cpp]: Oh.

File: source/text/Format.cpp [hunk 3]
```diff
@@ -53,7 +53,7 @@ namespace {
 	//
 	// The getter() acts like ConditionsStore.Get(), providing condition values.
 	// These are passed through Format::Whatever(), and appended to the result.
-	void AppendCondition(string &result, const string &source, Format::ConditionGetter getter,
+	void AppendCondition(string &result, const string &source, const Format::ConditionGetter& getter,
```
Comment (RisingLeaf) [inline source/text/Format.cpp:56]:
```suggestion
	void AppendCondition(string &result, const string &source, const Format::ConditionGetter &getter,
```

File: source/text/Format.cpp [hunk 4]
```diff
@@ -428,7 +428,7 @@ vector<string> Format::Split(const string &str, const string &separator)
 
 
 
-string Format::ExpandConditions(const string &source, ConditionGetter getter)
+string Format::ExpandConditions(const string &source, const ConditionGetter& getter)
```
Comment (RisingLeaf) [inline source/text/Format.cpp:431]:
```suggestion
string Format::ExpandConditions(const string &source, const ConditionGetter &getter)
```

File: source/text/DisplayText.cpp [hunk 5]
```diff
@@ -15,6 +15,8 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 #include "DisplayText.h"
 
+#include <utility>
+
```
Comment (RisingLeaf) [inline source/text/DisplayText.cpp:19]: why?

File: source/PlanetPanel.cpp [hunk 6]
```diff
@@ -344,7 +344,7 @@ void PlanetPanel::CheckWarningsAndTakeOff()
 		|| !leftUniques.empty())
 	{
 		ostringstream out;
-		auto ListUniques = [&out] (const map<const Outfit *, int> uniques)
+		auto ListUniques = [&out] (const map<const Outfit *, int>& uniques)
```
Comment (RisingLeaf) [inline source/PlanetPanel.cpp:347]:
```suggestion
		auto ListUniques = [&out] (const map<const Outfit *, int> &uniques)
```

File: source/ImageSet.h [hunk 7]
```diff
@@ -46,7 +46,7 @@ class ImageSet {
 public:
 	// ImageSets should be created with a name, as some image paths (e.g. plugin icons)
 	// do not contain the associated image name.
-	ImageSet(std::string name);
+	explicit ImageSet(std::string name);
```
Comment (RisingLeaf) [inline source/ImageSet.h:49]: const &?

File: source/EsUuid.cpp [hunk 8]
```diff
@@ -31,6 +31,7 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include <uuid/uuid.h>
 #endif
 
+
```
Comment (RisingLeaf) [inline source/EsUuid.cpp:34]:
```suggestion


```
even more!

File: source/ImageSet.h (hunk 7 above)
Comment (tibetiroka) [reply source/ImageSet.h:49]: I think you get the same number of copies if the constructor uses std::move, but I'm not sure...

File: source/EsUuid.cpp (hunk 8 above)
Comment (tibetiroka) [reply source/EsUuid.cpp:34]: My IDE is doing some funky stuff, apparently.

File: source/ImageSet.h (hunk 7 above)
Comment (tibetiroka) [reply source/ImageSet.h:49]:
```suggestion
	explicit ImageSet(const std::string &name);
```

File: source/ImageSet.cpp [hunk 9]
```diff
@@ -183,7 +183,7 @@ bool ImageSet::IsDeferred(const string &path)
 
 
 
-ImageSet::ImageSet(string name)
+ImageSet::ImageSet(const string &name)
 	: name(std::move(name))
```
Comment (warp-core) [inline source/ImageSet.cpp:187]: Again, if you're going to `std::move`, don't pass a `const &` in.

File: source/MapPanel.h [hunk 10]
```diff
@@ -184,7 +184,8 @@ class MapPanel : public Panel {
 	public:
 		Node(const Point &position, const Color &color, const std::string &name,
 			const Color &nameColor, const Government *government)
-			: position(position), color(color), name(name), nameColor(nameColor), government(government) {}
+			: position(std::move(position)), color(std::move(color)), name(std::move(name)),
+				nameColor(std::move(nameColor)), government(government) {}
```
Comment (warp-core) [inline source/MapPanel.h:188]: It also doesn't make sense to `std::move` from these const references.
Moving from a `Point` in general is just the same as copying from it anyway.
I'm fairly sure the same will be true for `Color`.

File: source/Messages.h [hunk 11]
```diff
@@ -40,7 +40,7 @@ class Messages {
 	public:
 		Entry() = default;
 		Entry(int step, const std::string &message, Importance importance)
-			: step(step), message(message), importance(importance) {}
+			: step(step), message(std::move(message)), importance(importance) {}
```
Comment (warp-core) [inline source/Messages.h:43]: And again.

File: source/PlayerInfoPanel.cpp [hunk 12]
```diff
@@ -916,6 +916,6 @@ PlayerInfoPanel::SortableColumn::SortableColumn(
 	Layout layout,
 	InfoPanelState::ShipComparator *shipSort
 )
-: name(name), offset(offset), endX(endX), layout(layout), shipSort(shipSort)
+: name(std::move(name)), offset(offset), endX(endX), layout(layout), shipSort(shipSort)
```
Comment (warp-core) [inline source/PlayerInfoPanel.cpp:919]:
```suggestion
	: name(std::move(name)), offset(offset), endX(endX), layout(layout), shipSort(shipSort)
```

File: source/Sprite.cpp [hunk 13]
```diff
@@ -59,7 +59,7 @@ namespace {
 
 
 Sprite::Sprite(const string &name)
-	: name(name)
+	: name(std::move(name))
```
Comment (warp-core) [inline source/Sprite.cpp:62]: Another one.

File: source/System.cpp [hunk 14]
```diff
@@ -49,7 +49,7 @@ const double System::DEFAULT_NEIGHBOR_DISTANCE = 100.;
 
 
 System::Asteroid::Asteroid(const string &name, int count, double energy)
-	: name(name), count(count), energy(energy)
+	: name(std::move(name)), count(count), energy(energy)
```
Comment (warp-core) [inline source/System.cpp:52]: And again.

File: source/text/Format.h [hunk 15]
```diff
@@ -71,7 +71,7 @@ class Format {
 	static std::vector<std::string> Split(const std::string &str, const std::string &separator);
 
 	// Finds &[condition] and &[format@condition] in strings and expands them
-	static std::string ExpandConditions(const std::string &source, ConditionGetter getter);
+	static std::string ExpandConditions(const std::string &source, const ConditionGetter& getter);
```
Comment (warp-core) [inline source/text/Format.h:74]:
```suggestion
	static std::string ExpandConditions(const std::string &source, const ConditionGetter &getter);
```

File: source/MapPanel.h (hunk 10 above)
Comment (tibetiroka) [reply source/MapPanel.h:188]: ~~why do we already have so many moves on Point then~~

File: source/Flotsam.cpp [hunk 16]
```diff
@@ -36,7 +36,7 @@ const int Flotsam::TONS_PER_BOX = 5;
 
 // Constructors for flotsam carrying either a commodity or an outfit.
 Flotsam::Flotsam(const string &commodity, int count, const Government *sourceGovernment)
-	: commodity(commodity), count(count), sourceGovernment(sourceGovernment)
+	: commodity(std::move(commodity)), count(count), sourceGovernment(sourceGovernment)
```
[the remark this answers is not in the document]
Comment (tibetiroka) [reply source/Flotsam.cpp:40]: Oh, this actually makes so much more sense than what my uni prof told me.

**Comment by tibetiroka** [comment]
I think I got them all. At least, in the files the static analyzers don't throw up on.

**Comment by warp-core** [comment]
While `move`ing from things like Points, Colors, and primitives is the same as copying, there is still value to leaving the `std::move`s there in those cases, especially in existing move constructors or assignment methods, for semantics, even if the result is no different.

**Comment by tibetiroka** [comment]
Sounds like something we should standardize, then, seeing how the current usage is a bit all over the place.

**Comment by tibetiroka** [comment]
I removed all of the `std::move` changes because there doesn't seem to be a guideline on when to use them in the code.

**Review by Amazinite — APPROVED** [review]

**Pull request merged**